common/str.h

```diff
@@ -217,6 +218,11 @@ class String {
 	 */
 	void trim();
 
+    /**
+     * Add a path component
+     */
+    String stringByAppendingPathComponent(String component, char sep = '/') const;
```

Comment:
This is definitiely nothing which should be in `Common::String` itself, it is a helper for filenames not a generic string helper. Since you only seem to use it in one place, I suggest to move it there (i.e. the chroot FS code).

But I think you don't even need this. As long as you can assure that `_root` is not empty, you can simply compose paths with: `Common::normalizePath(_root + '/' + component, '/');` in your case.